Add `Download Logs` option to Help and Support view

SanjulaGanepola opened the pull request:
### Changes

To help with gathering of logs, I added a new `Download Logs` option to the `Help and Support` view when connected to an IBM i. This will prompt users to select from different logs to download and the location where they would like to download it to. The files will be put into a zip with the following format: `CodeForIBMi_YYYYMMDD_HHMMSS.zip`. Note that if a log file is not found, the option will not be shown in the quick pick.

This should hopefully make it easier for users to share their logs in GitHub Issues instead of manually having to retrieve and copy them. Should also prevent the back and forth of explaining where to grab these logs :smiley:

Several new strings were added as well which will need to be translated.

![image](https://github.com/user-attachments/assets/f2b05239-1c67-4ac4-9d50-dafea586597b)

### How to test this PR

Examples:

1. Connect to an IBM i.
2. Click the `Download Logs` option.
3. Select the logs you want to download and then where you would like to download it to.
4. Verify the logs are downloaded.
5. Disconnect and verify the option is no longer visible.

### Checklist

* [x] have tested my change
* [x] Remove any/all `console.log`s I added

sebjulliand commented on src/views/helpView.ts:
```diff
@@ -93,6 +128,112 @@ async function openNewIssue() {
+        });
+      }
+
+      try {
+        const debugServiceLogPath = '/QIBM/UserData/IBMIDEBUGSERVICE/DebugService_log.txt';
+        const debugServiceLog = (await content.downloadStreamfileRaw(debugServiceLogPath));
+        if (debugServiceLog) {
+          logs.push({
+            label: t(`helpView.downloadLogs.debugServiceLog`),
+            detail: debugServiceLogPath,
+            picked: true,
+            fileName: 'DebugService.txt',
+            fileContent: debugServiceLog
+          });
+        }
+      } catch (err) { }
+
+      try {
+        const debugServiceEclipseInstancePath = '/QIBM/UserData/IBMIDEBUGSERVICE/startDebugService_workspace/.metadata/.log';
+        const debugServiceEclipseInstanceLog = (await content.downloadStreamfileRaw(debugServiceEclipseInstancePath));
+        if (debugServiceEclipseInstanceLog) {
+          logs.push({
+            label: t(`helpView.downloadLogs.debugServiceEclipseInstanceLog`),
+            detail: debugServiceEclipseInstancePath,
+            picked: true,
+            fileName: 'DebugServiceEclipseInstance.txt',
+            fileContent: debugServiceEclipseInstanceLog
+          });
+        }
+      } catch (err) { }
+    });
```
Preferably load the debug service configuration to get the actual path to the working directory.
```suggestion
      const debugConfig = await new DebugConfiguration().load();
      try {
        const debugServiceLogPath = `${debugConfig.getRemoteServiceWorkDir()}/DebugService_log.txt`;
        const debugServiceLog = (await content.downloadStreamfileRaw(debugServiceLogPath));
        if (debugServiceLog) {
          logs.push({
            label: t(`helpView.downloadLogs.debugServiceLog`),
            detail: debugServiceLogPath,
            picked: true,
            fileName: 'DebugService.txt',
            fileContent: debugServiceLog
          });
        }
      } catch (err) { }

      try {
        const debugServiceEclipseInstancePath = `${debugConfig.getRemoteServiceWorkDir()}/startDebugService_workspace/.metadata/.log`;
        const debugServiceEclipseInstanceLog = (await content.downloadStreamfileRaw(debugServiceEclipseInstancePath));
        if (debugServiceEclipseInstanceLog) {
          logs.push({
            label: t(`helpView.downloadLogs.debugServiceEclipseInstanceLog`),
            detail: debugServiceEclipseInstancePath,
            picked: true,
            fileName: 'DebugServiceEclipseInstance.txt',
            fileContent: debugServiceEclipseInstanceLog
          });
        }
      } catch (err) { }
    });
```

sebjulliand reviewed (changes requested):
Works fine, great idea too! Gathering the debug service log files will certainly be helpful 😄 
See my suggestion to get the actual path to the debug service work dir.

Also, two remarks:
- Maybe clear `outputChannelContent` in `IBMi::determineClear` method; it may grow up in size a bit too much if never cleared.
- Add placeholders in the language files for other to translate

I'll add the French translation.

sebjulliand commented on src/views/helpView.ts:
```diff
@@ -93,6 +128,112 @@ async function openNewIssue() {
+              await vscode.window.showInformationMessage(t(`helpView.downloadLogs.failed`, zipFile));
+            }
+          } catch (error) {
+            await vscode.window.showInformationMessage(t(`helpView.downloadLogs.failed`, zipFile));
```
Add `error` to the notification and show an error message instead of an information.

sebjulliand commented on src/views/helpView.ts:
```diff
@@ -93,6 +128,112 @@ async function openNewIssue() {
+                vscode.commands.executeCommand('revealFileInOS', vscode.Uri.file(downloadLocation))
+              }
+            } else {
+              await vscode.window.showInformationMessage(t(`helpView.downloadLogs.failed`, zipFile));
```
Show an error message instead of an information.

SanjulaGanepola commented:
@sebjulliand Addressed your comments and added placeholders in the other language files (38219fd). I left these placeholders as blank strings so let me know if this is fine or if I should put in the English ones as a placeholder for now.

Also, should there be any other log files we should be collecting? `/QIBM/ProdData/IBMiDebugService/bin/DebugService.env` or even the DAP trace?

sebjulliand reviewed (approved):
Great job @SanjulaGanepola !
Let's leave the translation files like that, it will be easier for others to spot where translations are needed.
As for the debug service DAP, let's not get them, at least for now. They seem to never be cleaned up so there can be a lot of them at some point.

This PR is approved, feel free to merge if you're fine with it! This is a very nice addition :)

SanjulaGanepola commented:
Everything looks good so it can be merged now. I don't have permissions to do that so can you merge it. Thanks

sebjulliand commented:
> Everything looks good so it can be merged now. I don't have permissions to do that so can you merge it. Thanks

Oh sure, sorry about that 😅
